Framework/src/TaskRunner.cxx

```diff
@@ -421,16 +423,36 @@ void TaskRunner::finishCycle(DataAllocator& outputs)
   }
 }
 
+void TaskRunner::updateMonitoringStats(ProcessingContext& pCtx)
+{
+  mNumberMessagesReceivedInCycle++;
+  for (const auto& input : InputRecordWalker(pCtx.inputs())) {
+    if (input.header != nullptr) {
```

Comment:
InputRecordWalker walks only on non-empty headers anyway, no need for that check.